chapters/annotations.tex

```diff
@@ -68,27 +73,185 @@ \section{Annotations for Documentation}\doublelabel{annotations-for-documentatio
+class may define multiple \lstinline!preferred! figures.
+
+The \lstinline!caption! attribute of \lstinline!Figure! can use the restricted
+form of text markup described in \autoref{text-markup-in-captions} as well as
+the variable replacements described in \autoref{variable-replacements}.
+
+\subsubsection{Plot Curves}\doublelabel{plot-curves}
+The actual data to plot is specified in the \lstinline!curves! of a \lstinline!Plot!:
+\begin{lstlisting}[language=modelica]
+record Curve
+  expression x = time "X coordinate values";
+  expression y "Y coordinate values";
+  String legend "Legend";
+end Curve;
 \end{lstlisting}
 
-The \lstinline!preferredView! annotation defines the default view when selecting the
-class. \lstinline!info! means info layer, i.e., the documentation of the class,
-\lstinline!diagram! means diagram layer and \lstinline!text! means the Modelica text layer.
-\begin{lstlisting}[language=grammar]
-documentation-class-annotation:
-   annotation "(" DocumentationClass "=" true ")"
+The mandatory \lstinline!x! and \lstinline!y! expressions are restricted to be
+component references refering to a scalar variable or \lstinline!time!.
+
+When \lstinline!legend! isn't provided, the tool produces a default based on
+\lstinline!x! and/or \lstinline!y!.  Providing the empty string as
+\lstinline!legend! means that the curve shall be omitted from the plot legend.
+Variable replacements, as described in \autoref{variable-replacements}, can be
+used in the \lstinline!legend! of \lstinline!Curve!
+
+\subsubsection{Axis Properties}\doublelabel{axis-properties}
+Properties may be defined for each \lstinline!Plot! axis:
+\begin{lstlisting}[language=modelica]
+record Axis
+  Real min "Axis lower bound";
+  Real max "Axis upper bound";
+  String unit "Unit of min and max";
+  String label "Axis label";
+end Axis;
 \end{lstlisting}
 
-Only allowed as class annotation on any kind of class and implies that
-this class and all classes within it are treated as having the
-annotation preferredView="info". If the annotation preferredView is
-explicitly set for a class, it has precedence over a DocumentationClass
-annotation \emph{{[}A tool may display such classes in special ways. For
-example, the description texts of the classes might be displayed instead
-of the class names, and if no icon is defined, a special information
-default icon may be displayed in the package browser.{]}}
+When an axis bound isn't provided, the tool computes one automatically.
+
+The Modelica tool is responsible for showing the unit used for values at the
+axis tick marks, so the axis \lstinline!label! shall not be used to convey this
+information.
+
+When an axis label isn't provided, the tool produces a default label. Providing
+the empty string as axis label means that no label should be shown. Variable
+replacements, as described in \autoref{variable-replacements}, can be used in
+the \lstinline!label! of \lstinline!Axis!
+
```

Comment:
This could be a good place to insert a section about the inheritance of figures.  Since inheritance of annotations isn't a well defined concept, it probably needs a sentence or two of explanation.  Ideally, the way we write this should also be coordinated with how we explain the inheritance of `experiment` annotations (decided, but not yet formulated).

Reply:
I have made the conclusion that they are completely orthogonal. Here, we don't need any merging/overriding/etc but there can exist a `figures` annotation on any level in the inheritance hierarchy and they live their own life. However, maybe we should make it explicit that you are expected to access (show) Figures from inherited classes as well?

Reply:
Yes.  Perhaps #2544 should be formulated more in terms of inheritance?  Perhaps something like this:
> The `figures` annotation is inherited in the sense that each class has a collection of figures comprised by the contents of the `figures` annotation in the class itself, as well the `figures` annotations from any base classes.

Then, the uniqueness requirement on the `Figure` `identifier` follows naturally.

It might also be worth noting that there could be a difference between referring to (from the `info`) an inherited figure via the present class versus via a parent class.  The difference might not be clear now, but it would probably make a difference when thinking of what simulation result to use when rendering figures inline in the class documentation.

Reply:
Just so we get somewhere here, I made a PR for this: https://github.com/modelica/ModelicaSpecification/pull/2588

Reply:
The PR #2588 has been merged.  Resolving conversation.